03-first-task.Rmd

```diff
@@ -2,4 +2,66 @@
+| disks          |                                                                                              |                                   |                                   |                                                                                             |
+| docker         | Run the task in this Docker image                                                            | Run the task in this Docker image | Run the task in this Docker image | Run the task in this Docker image                                                           |
+| maxRetries     |                                                                                              |                                   |                                   |                                                                                             |
+| memory         | Maximum amount of memory to use                                                              | n/a                               |                                   | Minimum amount of memory to use (scales with CPU)                                           |
+| preemptible    | n/a                                                                                          | n/a                               | n/a                               | Attempt running on a preemptible instance this many times, then switch to a non-preemptible |
+| walltime       | How much  [walltime](https://en.wikipedia.org/wiki/Elapsed_real_time)  to request for a task | n/a                               | n/a                               | n/a                                                                                         |
+
+
+
+
+## Command section 
+The command section of a WDL task is a bash script that will be run by the WDL executor. If a `docker` runtime attribute is specified, it will execute this bash script in that Docker image, otherwise, it will generally execute the bash script in a generic Ubuntu image.
+
+### Heredoc-style syntax versus braces syntax
+You may see WDLs that use this notation for the command section in a task:
+
+task do_something {
+  input {
+    String some_string
+  }
+  command {
+    some_other_string="FOO"
+    echo ${some_string}
+    echo $some_other_string
+  }
+}
+
+We recommend using heredoc-style syntax instead:
+task do_something {
+  input {
+    String some_string
+  }
+  command <<<
+    some_other_string="FOO"
+    echo ~{some_string}
+    echo $some_other_string
+  >>>
+}
+
+Heredoc-style syntax for command sections can be clearer than the alternative, as it makes a clearer distinction between bash variables and WDL variables. This is especially helpful for complicated bash scripts. Heredoc-style syntax is also what the WDL 1.1 spec recommends using in most cases. However, the older non-heredoc style is still perfectly functional for a lot of use cases.
+
+
+
+### `sep=` versus `sep()`
```

Comment:
will try to make a 1.1 / 1.0 toggle section, if not, move elsewhere

Reply:
on second thoughts, this is really the only one that's really a glaring difference at the moment. this should be a warning block.